test/ssltest_old.c

```diff
@@ -781,7 +781,7 @@ static void print_details(SSL *c_ssl, const char *prefix)
                prefix,
                SSL_get_version(c_ssl),
                SSL_CIPHER_get_version(ciph), SSL_CIPHER_get_name(ciph));
-    cert = SSL_get_peer_certificate(c_ssl);
+    cert = SSL_get1_peer_certificate(c_ssl);
```

Comment:
```suggestion
    cert = SSL_get0_peer_certificate(c_ssl);
```
And then delete the `X509_free(cert)` below.